### PR TITLE
IEP-264: Flash the binaries before starting the JTAG debugging

### PR DESCRIPTION
**Pull request by sigmaaa**
Starting the debugging session automatically flashes the program over JTAG. Tested only on macOS yet, but should work on Windows and Linux also. 
The user is now able to debug a program, which was not built and flashed before starting the debug session.
Use cases:

- create a new project > create a ESP-IDF debug configuration > go to debugger window and select necessary options > finish > launch in "Debug" mode 
Expected result: program will build and flash first, then will be started the debugging session.

- create a new project > build the project > create a ESP-IDF debug configuration > go to debugger window and select necessary options > finish > launch in "Debug" mode 
Expected result: the program will flash first, then will be started the debugging session.

### CODE REVIEW

File: bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/Configuration.java [hunk 1]
```diff
@@ -99,13 +102,18 @@ public static String getGdbServerCommandLine(ILaunchConfiguration configuration)
...
-
 			other = resolveAll(other, configuration);
 
 			if (other != null && !other.isEmpty()) {
 				lst.addAll(StringUtils.splitCommandLineOptions(other));
 			}
 
+			if (checkIfJtagIsAvailable())
+			{
+				lst.add("-c");
```
Comment (kolipakakondal) [inline bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/Configuration.java:113]: Add Non-NLS

File: bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/Configuration.java [hunk 2]
```diff
@@ -119,6 +127,38 @@ public static String getGdbServerCommandLine(ILaunchConfiguration configuration)
...
+	private static boolean checkIfJtagIsAvailable()
+	{
+		EspConfigParser parser = new EspConfigParser();
+		String openOCDPath = new IDFEnvironmentVariables().getEnvValue(IDFEnvironmentVariables.OPENOCD_SCRIPTS);
+		if (!openOCDPath.isEmpty() && parser.hasBoardConfigJson())
+		{
+			return true;
+		}
+		return false;
+	}
```
Comment (kolipakakondal) [inline bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/Configuration.java:139]: Can you extract this code and use it in both the places https://github.com/espressif/idf-eclipse-plugin/blob/b00dde41824a1df851ce0df17f4493e1ce89e9a2/bundles/com.espressif.idf.launch.serial.ui/src/com/espressif/idf/launch/serial/ui/internal/CMakeMainTab2.java#L164

Comment (kolipakakondal) [reply bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/Configuration.java:139]: EspFlashCommandGenerator can we renamed to ESPFlashUtil and move this method here?

File: bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/Configuration.java [hunk 3]
```diff
@@ -119,6 +127,38 @@ public static String getGdbServerCommandLine(ILaunchConfiguration configuration)
...
+			}
+			espFlashCommand = espFlashCommand.replace("<path-to-build-dir>", buildPath); //$NON-NLS-1$
+		}
+		catch (CoreException e)
+		{
+			Logger.log(e);
+		}
+		return espFlashCommand;
+	}
+
```
Comment (kolipakakondal) [inline bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/Configuration.java:161]: I think this was also used during JTAG flashing implementation, can we move this to a common place?

**Review by kolipakakondal — CHANGES_REQUESTED** [review]
Thanks for the PR @sigmaaa, could you please work on the review comments.

File: bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/Configuration.java (hunk 1 above)
Comment (sigmaaa) [reply bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/Configuration.java]: @kolipakakondal replaced with ESPFlashUtil.getEspJtagFlashCommand method

File: bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/Configuration.java (hunk 2 above)
Comment (sigmaaa) [reply bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/Configuration.java]: @kolipakakondal good point, thanks. Fixed

File: bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/Configuration.java (hunk 3 above)
Comment (sigmaaa) [reply bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/Configuration.java]: @kolipakakondal moved to ESPFlashUtil

**Review by kolipakakondal — APPROVED** [review]
LGTM

**Pull request merged**